pkg/pgmodel/pgx.go

```diff
@@ -391,14 +440,14 @@ func (p *pgxInserter) metricTableInsert(metric string, input chan insertDataRequ
 			return
 		}
 		needsResponse = append(needsResponse, insertDataTask{finished: req.finished, errChan: req.errChan})
-		batch.sampleInfos = append(batch.sampleInfos, req.data.sampleInfos...)
+		batch.sampleInfos = append(batch.sampleInfos, req.data...)
```

Comment:
This should call batch.AppendSlice (new function) or better yet batch.Append should be changed to work with this.